### PR TITLE
Deprecate Python Native API

### PR DESCRIPTION
**Pull request by tanwarsh**
- Marked the Python Native API as deprecated in all docs
- Moved the tutorials based on this API under deprecated/native_api
- Removed dependency on python native api code ([openfl-workspace/torch_llm_horovod/src/InHorovodrun.py](https://github.com/securefederatedai/openfl/compare/develop...tanwarsh:openfl:19-Nov-24?expand=1#diff-869591ea31812a4fc5392bb3f06fe5c0df76339e3d67a560f6f06108f0482203)).

### CODE REVIEW

**Review by rahulga1 — APPROVED** [review]

**Review by rajithkrishnegowda — APPROVED** [review]

File: openfl-workspace/torch_llm_horovod/src/InHorovodrun.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def get_args():
 
 def main():
     logger = getLogger(__name__)
-    fx.setup_logging(level="INFO", log_file=None)
+    setup_logging
```
Comment (theakshaypant) [inline openfl-workspace/torch_llm_horovod/src/InHorovodrun.py]: Do the 2 functions have the exact same functionality?
Also is the `()` not required here?

**Review by theakshaypant — APPROVED** [review]
LGTM, just a small question.

File: openfl-workspace/torch_llm_horovod/src/InHorovodrun.py (hunk 1 above)
Comment (tanwarsh) [reply openfl-workspace/torch_llm_horovod/src/InHorovodrun.py]: Thanks for pointing this out. I missed adding (). and yes the functionality is same for both function apart from importing tensorflow in native api which is not required here.

File: openfl-workspace/torch_llm_horovod/src/InHorovodrun.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 import horovod.torch as hvd
 
-import openfl.native as fx
```
Comment (teoparvanov) [inline openfl-workspace/torch_llm_horovod/src/InHorovodrun.py:12]: Why is this change of `InHorovodrun.py` needed, given that at this stage we're just deprecating the Python API?

**Review by teoparvanov — APPROVED** [review]
LGTM!

**Pull request merged**